File: patterns/c-patterns/pattern24.c
// Pattern 24: ...

// ****A
// ***BB
// **CCC
// *DDDD
// EEEEE

#include <stdio.h>
int main()
{
    int i, j;
    int n = 4;
    for (i = 0; i <= n; i++)
    {
        for (j = n; j >= 0; j--)
        {
            if (j > i)
                printf("*");
            else
                printf("%c", i + 65);
        }
        printf("\n");
    }
    return 0;
}

File: patterns/javascript-patterns/pattern24.js
// Pattern 24: ...

// ****A
// ***BB
// **CCC
// *DDDD
// EEEEE

const n = 4;

for (let i = 0; i <= n; i++) {
    let row = '';

    for (let j = n; j >= 0; j--) {
        if (j > i) {
            row += '*';
        } else {
            row += String.fromCharCode(i + 65);
        }
    }

    console.log(row);
}